src/App.tsx

```diff
@@ -1,14 +1,56 @@
 /* eslint-disable max-len */
-import React from 'react';
+import {
+  useCallback,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
-import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+import { TodoModal } from './components/TodoModal';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isLoadTodos, setIsLoadTodos] = useState(false);
+  const [todoId, setTodoId] = useState(0);
+  const [status, setStatus] = useState('all');
+  const [query, setQuery] = useState('');
+
+  const loadTodos = useCallback(async () => {
+    const dataTodos = await getTodos();
+
+    setTodos(dataTodos);
+    setIsLoadTodos(true);
+  }, []);
+
+  useEffect(() => {
+    loadTodos();
+  }, []);
+
+  const getCheckQuery = useCallback((str: string) => {
+    return str.toUpperCase().includes(query.toUpperCase().trim());
+  }, [query]);
+
+  const filteredTodos = useMemo(() => {
+    return todos.filter((todo) => {
+      if (status === 'active') {
```

Comment:
It would be better to use enum values instead of strings 'active', 'completed' for sort type

Also, use a `switch/case` here